rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -975,4 +998,59 @@ private int getParametersLength(String[] parameters) {
         }
     }
 
+    private Map<String, Object> processHistoryResult(String mbeanHistory, String range) throws IOException {
+        mbeanHistory = removingInternalEscaping(mbeanHistory);
+        Map<String, Object> jsonMap = parseJSON(mbeanHistory);
+        Map<String, Object> processedJsonMap = new HashMap<>();
+        if (jsonMap == null) {
+            return null;
+        }
+        long currentTime = new Date().getTime() / 1000;
+        long duration = StatHistory.Range.valueOf(range).getDuration();
+        int size = Optional.ofNullable(((ArrayList) jsonMap.entrySet().iterator().next().getValue()).size()).orElse(0);
+        long step = duration / size;
+
+        List<String> labels = new ArrayList<>();
+        for (int i = 0; i <= size; i++) {
+            //The time instant where the metric is taken
+            long t = currentTime - duration + step * i;
+            Date date = new Date(t * 1000);
+
+            labels.add(DATE_FORMAT.format(date));
+        }
+        JSONObject measureInfo;
+        JSONArray measureInfoList;
+        for (String key : jsonMap.keySet()) {
+            ArrayList values = (ArrayList) jsonMap.get(key);
+            measureInfoList = new JSONArray();
+            for (int i = 0; i < size; i++) {
+                measureInfo = new JSONObject();
+                measureInfo.put("value", Optional.ofNullable(values.get(i)).orElse(0));
+                measureInfo.put("from", labels.get(i));
+                measureInfo.put("to", labels.get(i + 1));
+                measureInfoList.add(measureInfo);
+            }
+            processedJsonMap.put(key, measureInfoList);
+        }
+        return processedJsonMap;
+    }
+
+    private String removingInternalEscaping(String result) {
+        result = result.replace("\\\"", "\"");
+        result = result.replace("\"{", "{");
+        result = result.replace("}\"", "}");
+        return result;
+    }
+
+    private Map<String, Object> parseJSON(final String jsonString) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            final JsonParser parser = mapper.getJsonFactory().createJsonParser(jsonString);
```

Comment:
Yes, these methods are deprecated since 2.1 version. However, we are using an old version (build.gradle of scheduler-api project).      
compile group: 'org.codehaus.jackson', name: 'jackson-mapper-asl', version: '1.9.12'
Should I upgrade the used version ? or import  the new version in the build.gradle of the rest-server project.

Reply:
Jackson libraries are doing things well. They already implemented other method that do the same job. I show you an example in my upper comment how you could rewrite the code with methods that are not deprecated.